engine/gui/widgets.py

```diff
@@ -89,10 +91,83 @@ def create(
                 None,
             )
 
-        return GUISpec(
+        gui_spec = GUISpec(
             assets=[Asset(**a) for a in spec.get("assets", [])],
             buttons=buttons,
             images=[Image(**b) for b in spec.get("images", [])],
             cancel_action=cancel_action,
             initial_selected_button=selected_button,
         )
+
+        cls.validate(gui_spec)
+        return gui_spec
+
+    def validate(self) -> None:
+        """Validates the GUISpec."""
+
+        if not self.buttons and not self.images:
+            raise ValidationError("Must have at least one button or image")
+
+        asset_names = collections.Counter([a.name for a in self.assets])
+        duplicate_assets = [asset for asset, count in asset_names.items() if count > 1]
+        if duplicate_assets:
+            raise ValidationError(f"Duplicate asset name: {duplicate_assets}.")
+
+        button_names = collections.Counter([b.name for b in self.buttons])
+        duplicate_buttons = [name for name, count in button_names.items() if count > 1]
+        if duplicate_buttons:
+            raise ValidationError(f"Duplicate button name: {duplicate_buttons}")
+
+        image_names = collections.Counter([i.image_asset for i in self.images])
+        duplicate_images = [image for image, count in image_names.items() if count > 1]
+        if duplicate_images:
+            raise ValidationError(f"Duplicate image name(s): {duplicate_images}")
+
+        for b_names in button_names.items():
+            if b_names[0] not in list(asset_names.elements()):
+                raise ValidationError(f"Button {b_names[0]} not in assets")
```

Comment:
This check doesn't quite do what you think it does: it's checking to see if the name of the button is in assets. This is not correct; buttons can have whatever names they like and they don't have to be connected to assets. Instead it's the asset names within the button definition that must exist as assets.

For example, here's a spec that is valid but would fail this validation check (note that the name of the button does not match the name of the asset, but the asset is being referred to by the button):
```json
{
  "assets": [
    {
      "name": "some-asset",
      "path": "/some/path"
    }
  ],
  "buttons": [
    {
      "name": "my-button",
      "selected_image_asset": "some-asset"
    }
  ]
}
```
Conversely, here's a spec that is invalid but would pass (because the button named `something` is referring to an asset called `nonexistent-asset`, which doesn't exist):
```json
{
  "assets": [
    {
      "name": "something",
      "path": "/some/path"
    }
  ],
  "buttons": [
    {
      "name": "something",
      "selected_image_asset": "nonexistent-asset"
    }
  ]
}
```